tfjs-layers/src/engine/container.ts

```diff
@@ -594,13 +594,37 @@ export abstract class Container extends Layer {
   loadWeights(weights: NamedTensorMap, strict = true) {
     const nameToWeight: {[name: string]: LayerVariable} = {};
     let totalWeightsCount = 0;
-    for (const layer of this.layers) {
-      for (const weight of layer.weights) {
-        if (nameToWeight[weight.originalName] != null) {
-          throw new ValueError(`Duplicate weight name: ${weight.originalName}`);
+    // get weights key from tensor map in order to check if it is from keras v3.
+    // e.g. dense/0
+    const key = Object.keys(weights)[0].split('/');
+
+    // Check if weights from keras v3.
+    if (this.isV3(key[key.length - 1])) {
+      for (const layer of this.layers) {
+        layer.weights.forEach((weight, index) => {
```

Comment:
Since the most codes of the two branches are same, we could merge the two branch and move the `isV3` check in the middle:
```
const isV3Model = this.isV3(key[key.length - 1]);
...
    const parsedName = isV3 ? 
                 weight.name.split('/').slice(0, -1).join('/') + '/' + index : weight.name;
...
```